include/mata/nfa.hh

```diff
@@ -645,7 +600,7 @@ public:
      */
     void unify_final();
 
-    bool is_state(const State &state_to_check) const { return state_to_check < delta.size(); }
+    bool is_state(const State &state_to_check) const { return state_to_check <= delta.max_state(); }
```

Comment:
Should we ask the automaton about his max_state rather than delta?

Reply:
this makes sense in the current implementation. Need to check whether it does not break anything.

Reply:
solved.